nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -24,6 +24,26 @@ let
           '';
         };
 
+        devices = mkOption {
+          type = with types; listOf str;
```

Comment:
Is there a particular reason to prefer magic strings for `devices` over a `fileSystems`-style attrset? It'd make the source/destination far more obvious  for one, and also make duplication/conflicts obvious at the NixOS level.

Reply:
I've mostly done it this way as the `volumes` and `ports` options also chose this colon:delimited, docker-esque way. I'd probably agree that the `fileSystems`-esque nixy approach makes more sense for our uses, but if we're going to do that, then we'll have to also implement the same concept for those options as well.

Reply:
I'm not particularly against migrating those to a more sensible scheme, and I'd be happy to put the work in for `volumes`. `ports` includes ranges and is a somewhat more ambiguous case, though matching `networking.firewall` in that respect wouldn't be a bad option.